ExtractAWData/ExtractAWData.cs

```diff
@@ -0,0 +1,61 @@
+using Azure.Storage.Blobs;
+using Gwa.Etl.Models;
+using Gwa.Etl.Services;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Gwa.Etl
+{
+    public class ExtractAWData
+    {
+        private readonly BlobClient blobClient;
+        private readonly IConfiguration configuration;
+        private readonly IHttpClientFactory httpClientFactory;
+        private readonly ILogger<ExtractAWData> logger;
+
+        public ExtractAWData(BlobClient blobClient, IConfiguration configuration, IHttpClientFactory httpClientFactory, ILogger<ExtractAWData> logger)
+        {
+            this.blobClient = blobClient;
+            this.configuration = configuration;
+            this.httpClientFactory = httpClientFactory;
+            this.logger = logger;
+        }
+
+        [Function("ExtractAWData")]
+        public async Task Run([TimerTrigger("0 0 8 * * 0")] TimerInfo timerInfo)
+        {
+            try
+            {
+                logger.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}.");
+                logger.LogInformation($"Last execution was at {timerInfo.ScheduleStatus.Last}. Next execution will be at {timerInfo.ScheduleStatus.Next}");
+
+                AirWatchService getUsers = new(configuration, httpClientFactory, logger);
```

Comment:
The shorthand `new` syntax isn't typically used in other parts of Defra with people favouring `var`, i.e.
`var getUsers = new AirWatchService(configuration, httpClientFactory, logger);`
There's benefits of each and no official guidelines in place on which approach to use, but worth noting what other teams are doing.